yt_dlp/extractor/instagram.py

```diff
@@ -360,49 +360,83 @@ def _extract_embed_urls(cls, url, webpage):
 
     def _real_extract(self, url):
         video_id, url = self._match_valid_url(url).group('id', 'url')
-        general_info = self._download_json(
-            f'https://www.instagram.com/graphql/query/?query_hash=9f8827793ef34641b2fb195d4d41151c'
-            f'&variables=%7B"shortcode":"{video_id}",'
-            '"parent_comment_count":10,"has_threaded_comments":true}', video_id, fatal=False, errnote=False,
-            headers={
-                'Accept': '*',
+        media = {}
+        webpage = {}
+        api_check = self._download_json(
+            f'https://i.instagram.com/api/v1/web/get_ruling_for_content/?content_type=MEDIA&target_id={_id_to_pk(video_id)}',
+            video_id, fatal=False, note='Setting up session', errnote=False, headers={
+                'Accept': '*/*',
                 'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36',
```

Comment:
for authenticated users, it's preferable to maintain a constant UA in case of instagram http fingerprinting to avoid the user's account being flagged as suspicious. ideally, logged-in users would pass their real UA via the --user-agent option, but leaving hardcoded for now and changing to Windows Chrome UA